pkg/minikube/cruntime/docker.go

```diff
@@ -688,6 +688,14 @@ const (
 	CNICacheDir = "/var/lib/cni/cache"
 )
 
+func getCriDockerdPath(cr CommandRunner) string {
+        rr, err := cr.RunCmd(exec.Command("which", "cri-dockerd"))
```

Comment:
`which` is not a standard command, and parsing its output is dangerous at best. The stdlib [`LookPath`](https://pkg.go.dev/os/exec#LookPath) is designed for this use-case.

Reply:
How do you run LookPath over ssh ? We could run `command -v`, at least for the recently modern ones

Reply:
Hmm, `command -v` is at least POSIX. I suppose the remote use-case makes this harder; but at the same time it feels very impure. The ability to `go run` over SSH might be nice in the end, but that's a lot of machinery to build out.

Does it make sense to abstract this a layer deeper and write a `cr.LookPath` that uses the external binary or `os.LookPath` based on the target?

Reply:
Sure, but that is more of a separate refactor